components/file_preview_modal/image_preview.tsx

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef, useMemo, useEffect, useState} from 'react';
+import {clamp} from 'lodash';
+
+import {getFilePreviewUrl, getFileDownloadUrl} from 'mattermost-redux/utils/file_utils';
+
+import './image_preview.scss';
+import {FileInfo} from '@mattermost/types/files';
+
+import {ZoomValue} from './file_preview_modal_image_controls/file_preview_modal_image_controls';
+import {LinkInfo} from './types';
+
+const HORIZONTAL_PADDING = 48;
+const VERTICAL_PADDING = 168;
+
+const SCROLL_SENSITIVITY = 0.003;
+const MAX_ZOOM = 5;
+
+let zoom: number;
+let minZoom: number;
```

Comment:
same thing applies here: please use a `ref` when you need variables to be mutated and not in state

Reply:
Would I still be able to export them?

Reply:
@michelengelen I can't figure out a way to use ref while still exporting them..

Reply:
I tried something but I'm not sure it's ideal